pkg/server/api/agent/v1/service.go

```diff
@@ -387,6 +387,20 @@ func (s *Service) RenewAgent(ctx context.Context, req *agentv1.RenewAgentRequest
 		return nil, api.MakeErr(log, codes.Internal, "caller ID missing from request context", nil)
 	}
 
+	attestedNode, err := s.ds.FetchAttestedNode(ctx, callerID.String())
```

Comment:
can you add test cases:
- fetchAttestedNode return error
- fetchAttestedNode not found

Reply:
These test cases already exist from the call to `updateAttestedNode()` below. I updated the error message in the tests to reflect it will fail `FetchAttestedNode()` instead.